datafusion/execution/src/cache/cache_manager.rs

```diff
@@ -189,13 +228,20 @@ pub struct CacheManagerConfig {
     /// Avoid get same file statistics repeatedly in same datafusion session.
     /// Default is disable. Fow now only supports Parquet files.
     pub table_files_statistics_cache: Option<FileStatisticsCache>,
-    /// Enable cache of file metadata when listing files.
-    /// This setting avoids listing file meta of the same path repeatedly
-    /// in same session, which may be expensive in certain situations (e.g. remote object storage).
+    /// Enable caching of file metadata when listing files.
+    /// Enabling the cache avoids repeat list and metadata fetch operations, which may be expensive
+    /// in certain situations (e.g. remote object storage), for objects under paths that are
+    /// cached.
     /// Note that if this option is enabled, DataFusion will not see any updates to the underlying
-    /// location.  
-    /// Default is disable.
-    pub list_files_cache: Option<ListFilesCache>,
+    /// storage for at least `list_files_cache_ttl` duration.
+    /// Default is disabled.
+    pub list_files_cache: Option<Arc<dyn ListFilesCache>>,
+    /// Limit the number of objects to keep in the `list_files_cache`. Default: ~125k objects
+    pub list_files_cache_limit: usize,
```

Comment:
I found it a little strange that the cache size is set on the cache manager itself rather than an options struct -- though now i see it is consistent with how metadata_cache_limits work

Reply:
Yes, it does seem like this API could be easier to use. It seems to me that this would be something that we might want to do as future work and fix up both the metadata and list cache at the same time since this PR has already grown to a pretty sizeable amount of code.